testsuite/MDAnalysisTests/analysis/test_base.py

```diff
@@ -162,6 +162,22 @@ def test_AnalysisFromFunction(u, start, stop, step, nframes):
         assert_equal(results, ana.results)
 
 
+def test_AnalysisFromFunction_args_order(u):
+    # Issue 2503
+    def mass_xyz(atomgroup, masses):
+        return atomgroup.positions * masses
+    
+    protein = u.select_atoms('protein')
+    ans1 = base.AnalysisFromFunction(mass_xyz, u.trajectory, protein, protein.masses.reshape(-1, 1))
+    ans2 = base.AnalysisFromFunction(mass_xyz, protein, protein.masses.reshape(-1, 1))
```

Comment:
Please follow [PEP8](https://www.mdanalysis.org/UserGuide/contributing_code.html#code-formatting-in-python) line lengths when possible.